packages/moda-sdk/src/profile/updateProfileFor.ts

```diff
@@ -0,0 +1,45 @@
+import { Address, Hash } from 'viem';
+import { Config, ConfigError } from '../types';
+import { isValidAddress } from '../utils/isValidAddress';
+import { Result } from '../utils/types';
+import { verifyConfig } from '../utils/verifyConfig';
+import { ProfileABI } from './abi/ProfileABI';
+import { ProfileError } from './types';
+
+interface UpdateProfileForArgs {
+  config: Config;
+  profileAddress: Address;
+  contractAddress: Address;
+  metadataUri: string;
+}
+
+export async function updateProfileFor(
+  updateProfileForArgs: UpdateProfileForArgs,
+): Promise<Result<Hash, ProfileError | ConfigError>> {
+  const { config, profileAddress, contractAddress, metadataUri } =
+    updateProfileForArgs;
+
+  const verifiedConfig = verifyConfig(config);
+  if (!verifiedConfig.ok) return verifiedConfig;
+  const { publicClient, walletClient } = verifiedConfig.value;
+
+  if (!isValidAddress(profileAddress))
+    return { ok: false, error: ProfileError.AddressInvalid };
+
+  if (!isValidAddress(contractAddress))
+    return { ok: false, error: ProfileError.AddressInvalid };
+
+  const [account] = await walletClient.getAddresses();
+
+  const { request } = await publicClient.simulateContract({
+    account,
+    abi: ProfileABI,
+    address: profileAddress,
+    functionName: 'updateProfileFor',
+    args: [contractAddress, metadataUri],
+  });
+
+  const txHash = await walletClient.writeContract(request);
```

Comment:
What if this reverts? Will this blow up? Can we return an error instead? Let's do this for any write method in this contract